### PR TITLE
add many emscripten functions for PHP

### PR DESCRIPTION
**Pull request by MarkMcCaskey**

### CODE REVIEW

File: lib/emscripten/src/io/unix.rs [hunk 1]
```diff
@@ -15,3 +15,60 @@ pub fn printf(ctx: &mut Ctx, memory_offset: i32, extra: i32) -> i32 {
...
+    0
+}
+
+/// getprotobynumber
+pub fn getprotobynumber(_ctx: &mut Ctx, _one: i32) -> i32 {
+    debug!("emscripten::getprotobynumber");
+    0
+}
+
+// REVIEW: does this belong here?
```
Comment (xmclark) [inline lib/emscripten/src/io/unix.rs]: The location is fine. We can always move it later.

File: lib/emscripten/src/syscalls/unix.rs [hunk 2]
```diff
@@ -93,6 +105,91 @@ pub fn ___syscall5(ctx: &mut Ctx, _which: c_int, mut varargs: VarArgs) -> c_int
...
+/// getrusage
+pub fn ___syscall77(ctx: &mut Ctx, _which: c_int, mut varargs: VarArgs) -> c_int {
+    debug!("emscripten::___syscall77 (getrusage) {}", _which);
+
+    let resource: c_int = varargs.get(ctx);
+    let rusage_ptr: c_int = varargs.get(ctx);
+    #[allow(clippy::cast_ptr_alignment)]
+    let rusage = emscripten_memory_pointer!(ctx.memory(0), rusage_ptr) as *mut rusage;
+    // REVIEW: is this the correct way to verify alignment?
+    assert_eq!(8, mem::align_of_val(&rusage));
```
Comment (xmclark) [inline lib/emscripten/src/syscalls/unix.rs]: yeah this is fine

**Comment by MarkMcCaskey** [comment]
bors try

**Pull request merged**